CNV-47160: UDN support on public clouds (#88975)
openshift/openshift-docs: virt/vm_networking/virt-networking-overview.adoc

[NOTE]
====

> Connecting VMs to user-defined networks with the `layer2` topology is supported on public clouds.

Reviewer: I would keep the old note. And add this new one below it. Just replacing "supported" with "recommended"

You can expose a VM within the cluster or outside the cluster by creating a `Service` object.
endif::openshift-rosa,openshift-dedicated[]

[id="primary-udn-config"]
== Configuring a primary user-defined network

> xref:../../virt/vm_networking/virt-connecting-vm-to-primary-udn.adoc#virt-connecting-vm-to-primary-udn[Connecting a virtual machine to a primary user-defined network]::

Author: Similar formatting is used throughout this assembly. The link is a description list entry and points to the relevant Networking topic. I would like to keep it as is.

You can connect a virtual machine (VM) to a user-defined network (UDN) on the VM’s primary interface. The primary user-defined network replaces the default pod network to connect pods and VMs in selected namespaces. Tenant owners can configure a primary `UserDefinedNetwork` CRD to create a network that isolates their namespace from other namespaces without requiring network policies. Cluster administrators can use the `ClusterUserDefinedNetwork` CRD to create a shared OVN `layer2` network across multiple namespaces.
+

> User-defined networks with the `layer2` overlay topology are useful for VM workloads, and a good alternative to secondary networks in environments where physical network access is limited, such as the public cloud. The `layer2` topology enables seamless migration of VMs without the need for Network Address Translation (NAT) because persistent IP addresses are preserved across cluster nodes during live migration.

Reviewer: ```suggestion
User-defined networks with the `layer2` overlay topology are useful for VM workloads, and a good alternative to secondary networks in environments where physical network access is limited, such as the public cloud. The `layer2` topology enables seamless migration of VMs without the need for Network Address Translation (NAT) and it also provides persistent IP addresses preserved between reboots and during live migration.
```

> User-defined networks with the `layer2` overlay topology are a good fit for VM workloads, and a good alternative to secondary networks in environments where physical network access is limited, such as the public cloud. The `layer2` topology enables seamless migration of VMs without the need for Network Address Translation (NAT) because persistent IP addresses are preserved across cluster nodes during live migration.

Reviewer 2: "Good fit" seems like marketing parlance. How about:

```suggestion
User-defined networks with the `layer2` overlay topology are useful for VM workloads, and a good alternative to secondary networks in environments where physical network access is limited, such as the public cloud. The `layer2` topology enables seamless migration of VMs without the need for Network Address Translation (NAT) because persistent IP addresses are preserved across cluster nodes during live migration.
```
